File: frontend/src/models/post/PostDto.ts
import { TagDto } from "./TagDto"
import { FollowResponse } from "../user/UserResponse"

export type PostDto = {
     id: number
     title: string
     posterUrl: string
     description: string
     body: string
     tags: TagDto[]
     creator: FollowResponse
     views: number
     mark: number
     positiveMarks: number
     negativeMarks: number
     markStatus: boolean | null
     isMyProfileSubscribed: boolean | null
     createdAt: string
}

File: src/main/java/com/pwgp/blog/dto/post/PostDto.java
package com.pwgp.blog.dto.post;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.pwgp.blog.dto.user.UserResponse;
import lombok.Data;


import java.time.LocalDateTime;
import java.util.Set;


@Data
public class PostDto {

    private Long id;

    private String title;

    private String posterUrl;

    private String description;

    private String body;

    private Set<TagDto> tags;

    private UserResponse creator;

    private int views;

    private int mark;

    private int positiveMarks;

    private int negativeMarks;

    private Boolean markStatus;

    private Boolean isMyProfileSubscribed;

    private LocalDateTime createdAt;

}